test/extended/router/gatewayapicontroller.go

```diff
@@ -0,0 +1,274 @@
+
+			g.By("check OSSM Pod and istiod pod is present with OSSM Operator installation")
+			podList, err := coreClient.CoreV1().Pods("openshift-operators").List(context.Background(), metav1.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(podList).To(o.ContainSubstring(`servicemesh-operator3`))
+
+			podList, err = coreClient.CoreV1().Pods("openshift-ingress").List(context.Background(), metav1.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(podList).To(o.ContainSubstring(`istiod-openshift-gateway`))
+
+			g.By("Confirm that ISTIO CR is created and in healthy state")
+			resource := types.NamespacedName{Namespace: "openshift-ingress", Name: "openshift-gateway"}
+
+			istioStatus, err := oc.AsAdmin().Run("get").Args("-n", resource.Namespace, "istio", resource.Name, "-o=jsonpath={.status.state}").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(istioStatus).To(o.Equal("Healthy"))
+
+		})
+		g.It("and ensure custom gatewayclass can be accepted", func() {
+			gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
+			coreClient := clientset.NewForConfigOrDie(oc.AdminConfig())
+
+			g.By("Create Custom GatewayClass")
+			gatewayClass := buildGatewayClass("custom-gateway", gatewayClassControllerName)
+			gwc, err := gwapiClient.GatewayV1().GatewayClasses().Create(context.TODO(), gatewayClass, metav1.CreateOptions{})
+			if err != nil {
+				e2e.Logf("Gateway Class %s already exists, or has failed to be created, checking its status", "custom-gateway")
+			}
+			errCheck := checkGatewayClass(oc, "custom-gateway")
+			o.Expect(errCheck).NotTo(o.HaveOccurred())
+			e2e.Logf("GatewayClass %s successfully installed and accepted!", gwc.Name)
+
+			g.By("Deleting Custom GatewayClass and confirming that it is no longer there")
+			err = gwapiClient.GatewayV1().GatewayClasses().Delete(context.Background(), "custom-gateway", metav1.DeleteOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			_, err = gwapiClient.GatewayV1().GatewayClasses().Get(context.Background(), "custom-gateway", metav1.GetOptions{})
+			o.Expect(err).To(o.HaveOccurred())
+			e2e.Logf("The custom gatewayClass %s has been sucessfully deleted", "custom-gateway")
+
+			g.By("check if default gatewayClass is accepted and ISTIO CR and pod are still available")
+			defaultCheck := checkGatewayClass(oc, gatewayClassName)
+			o.Expect(defaultCheck).NotTo(o.HaveOccurred())
+
+			podList, err := coreClient.CoreV1().Pods("openshift-ingress").List(context.Background(), metav1.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(podList).To(o.ContainSubstring(`istiod-openshift-gateway`))
+
+			istioCR, err := oc.AsAdmin().Run("get").Args("-n", "openshift-ingress", "istio", "openshift-gateway").Output()
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(istioCR).To(o.ContainSubstring(`openshift-gateway`))
+		})
+
+		g.It("and HTTPRoute can be created and accepted", func() {
```

Comment:
@melvinjoseph86 just added the test for HTTPRoute for your reference, you could improve/refactor the code in your PR.